### ISSUE TITLE
padded-blocks fails on comment

### ISSUE THREAD

**Issue by keradus** [eslint/eslint#2788]
Hi. When I have a code:
```javascript
if (a) {
    // we really need to test it
    if (b) {
        c();
    }

    d();
}
```

The rule warns me: `Block must not be padded by blank lines  padded-blocks`.
It should not since comment is not blank line.

When I remove a comment then rule passes.

**Comment by lo1tuma** [comment]
Can you share your eslint configuration? Do you use the default parser?

**Comment by keradus** [comment]
```json
{
    "env":{
        "browser": true,
        "node": true
    },
    "rules": {
        // Possible Errors
        "comma-dangle": [2, "never"],
        "no-cond-assign": 2,
        "no-console": 0,
        "no-constant-condition": 0, // S3: docelowo 2
        "no-control-regex": 2, // S3: docelowo 2
        "no-debugger": 2,
        "no-dupe-args": 2,
        "no-dupe-keys": 2,
        "no-duplicate-case": 2,
        "no-empty-character-class": 2,
        "no-empty": 2,
        "no-ex-assign": 2,
        "no-extra-boolean-cast": 2,
        "no-extra-parens": [2, "functions"],
        "no-extra-semi": 2,
        "no-func-assign": 2,
        "no-inner-declarations": 2,
        "no-invalid-regexp": 2,
        "no-irregular-whitespace": 2,
        "no-negated-in-lhs": 2,
        "no-obj-calls": 2,
        "no-regex-spaces": 2,
        "no-reserved-keys": 2,
        "no-sparse-arrays": 2,
        "no-unreachable": 2,
        "use-isnan": 2,
        "valid-jsdoc": 2,
        "valid-typeof": 2,

        // Best Practices
        "accessor-pairs": 0,
        "block-scoped-var": 0,
        "complexity": 0,
        "consistent-return": 0,
        "curly": 2,
        "default-case": 0,
        "dot-notation": 0,
        "dot-location": [2, "object"], // S2: docelowo [2, "property"]
        "eqeqeq": 2,
        "guard-for-in": 2,
        "no-alert": 0,
        "no-caller": 2,
        "no-div-regex": 2,
        "no-else-return": 2,
        "no-empty-label": 2,
        "no-eq-null": 2,
        "no-eval": 2,
        "no-extend-native": 2,
        "no-extra-bind": 2,
        "no-fallthrough": 2,
        "no-floating-decimal": 2,
        "no-implied-eval": 2,
        "no-iterator": 2,
        "no-labels": 2,
        "no-lone-blocks": 2,
        "no-loop-func": 2,
        "no-multi-spaces": 0,
        "no-multi-str": 2,
        "no-native-reassign": 2,
        "no-new-func": 2,
        "no-new-wrappers": 2,
        "no-new": 2,
        "no-octal-escape": 2,
        "no-octal": 2,
        "no-param-reassign": 0, // S3: docelowo 2
        "no-process-env": 2,
        "no-proto": 2,
        "no-redeclare": 2,
        "no-return-assign": 2,
        "no-script-url": 2,
        "no-self-compare": 2,
        "no-sequences": 2,
        "no-throw-literal": 2,
        "no-unused-expressions": 2,
        "no-void": 2,
        "no-warning-comments": 0,
        "no-with": 2,
        "radix": 2,
        "vars-on-top": 2,
        "wrap-iife": 2,
        "yoda": 0,

        // Strict Mode
        "strict": 2,

        // Variables
        "no-catch-shadow": 2,
        "no-delete-var": 2,
        "no-label-var": 2,
        "no-shadow-restricted-names": 2,
        "no-shadow": 0,
        "no-undef-init": 2,
        "no-undef": 2,
        "no-undefined": 0,
        "no-unused-vars": 0,
        "no-use-before-define": 2,

        // Stylistic Issues
        "brace-style": [2, "stroustrup", { "allowSingleLine": true }],
        "camelcase": 0,
        "comma-spacing": 2,
        "comma-style": 0,
        "consistent-this": 0, // S3: docelowo [2, "self"]
        "eol-last": 2,
        "func-names": 0,
        "func-style": 2,
        "indent": [2, 4, {"indentSwitchCase": true}],
        "key-spacing": 0,
        "lines-around-comment": 0,
        "linebreak-style": 2,
        "max-nested-callbacks": 0,
        "new-cap": 2,
        "new-parens": 2,
        "newline-after-var": 2,
        "no-array-constructor": 2,
        "no-continue": 0,
        "no-inline-comments": 0,
        "no-lonely-if": 2,
        "no-mixed-spaces-and-tabs": 2,
        "no-multiple-empty-lines": 2,
        "no-nested-ternary": 0,
        "no-new-object": 2,
        "no-spaced-func": 0,
        "no-ternary": 0,
        "no-trailing-spaces": 2,
        "no-underscore-dangle": 0,
        "no-unneeded-ternary": 0,
        "no-wrap-func": 0, // deprecated, ale jeszcze jest domyślnie włączane a ma bugi - musimy ręcznie wyłączyć
        "object-curly-spacing": 0, // S2: docelowo [2, "always"]
        "one-var": 2,
        "operator-assignment": 2,
        "operator-linebreak": 2,
        "padded-blocks": 0, // S3: docelowo [2, "never"],
        "quote-props": [2, "as-needed"],
        "quotes": [2, "double"],
        "semi-spacing": 2,
        "semi": 2,
        "sort-vars": 0, // docelowo 2
        "space-after-keywords": 2,
        "space-before-blocks": 2,
        "space-before-function-paren": 2,
        "space-in-brackets": 0, // S2: docelowo [2, "always"]
        "space-in-parens": 2,
        "space-infix-ops": 2,
        "space-return-throw-case": 2,
        "space-unary-ops": 2,
        "spaced-line-comment": 2,
        "wrap-regex": 0
    }
}
```

**Comment by lo1tuma** [comment]
You disabled the rule in your configuration. So it should never warn.

However, I've tried your example in the [online demo](http://eslint.org/demo/) and can’t reproduce the problem:

```js
/* eslint padded-blocks: [2, "never"] */

if (a) {
    // we really need to test it
    if (b) {
        c();
    }

    d();
}
```

Which version of ESLint do you use?

**Comment by keradus** [comment]
` "padded-blocks": 0, // S3: docelowo [2, "never"],`
yeah, error occurs only with commented setting. I need to disable it due to that problem. Sorry for that.

ESLint version: 0.22.1

**Comment by keradus** [comment]
same on 0.23.0

**Comment by keradus** [comment]
Please try this
```js
/* eslint padded-blocks: [2, "never"] */

var xxx = function () {
    // foo
    if (
        // bar
        a ||
        // baz
        b
    ) {
        return;
    }
};
```

**Comment by lo1tuma** [comment]
Now I can reproduce it.

Here is more minimal example:

```js
/* eslint padded-blocks: [2, "never"] */

{
    // comment
    if (// comment
        a) {}
}
```

This is a problem with the comment attachment algorithm. Both comments getting attached to `Identifier` node of the if condition. The current implementation of the rule would expect that the first comment gets attached to the `IfStatement` node.

**Comment by nzakas** [comment]
Can you file an espree bug?

**Comment by moll** [comment]
Just stumbled upon this too. Same issue with trailing comments.
```javascript
function foo() {
  console.log("Hi")

  // Long text.
  // Long text.
  // Long text.
}
```

**Comment by lo1tuma** [comment]
@moll This is a different issue caused by the missing semicolon after `console.log("Hi")`, see #2336.

**Comment by lo1tuma** [comment]
For reference https://github.com/eslint/espree/issues/155

**Comment by keradus** [comment]
Great to have that fixed! Waiting for next stable release ;)
